refellips/structureSE.py

```diff
@@ -885,16 +885,21 @@ def overall_RI(slabs, solvent, ema="linear"):
 
         slabs[..., 1] += solvent.real**2 * vf
         slabs[..., 2] += solvent.imag**2 * vf
-    elif ema_method == "maxwell-garnett":
+    elif ema == "maxwell-garnett":
         slabs[..., 1:3] = slabs[..., 1:3] ** 2
 
         top_r = 2 * (1 - vf) * slabs[..., 1] + (1 + 2 * vf) * solvent.real**2
         bottom_r = (2 + vf) * slabs[..., 1] + (1 - vf) * solvent.real**2
         slabs[..., 1] *= top_r / bottom_r
 
-        top_r = 2 * (1 - vf) * slabs[..., 2] + (1 + 2 * vf) * solvent.imag**2
-        bottom_r = (2 + vf) * slabs[..., 2] + (1 - vf) * solvent.imag**2
-        slabs[..., 2] *= top_r / bottom_r
+        if (solvent.imag != 0) & (np.prod(slabs[..., 2]) != 0):
```

Comment:
Not sure about these lines here, let's talk B4 merge.

Reply:
No worries. The if statement on new line 895 was put there because if you use a Cauchy model, you have no EC (i.e., EC = 0) and you get a div 0 error which interrupts the calc. This was my quick fix.

Reply:
`np.prod` means that if any of the extinction coefficients are zero then this section doesn't run.

Reply:
Indeed - my bad. Okay if I replace this then with a try (everything in if statement), then except pass (or print("No EC")?